### PR TITLE
Added @example tags to src/table.js

### LINKED ISSUE #157: @example Comments should refer to sample code
nodejs-bigtable has `@example` comments that include code.  This is error prone, since it's not compiled and may get obsolete.  Instead of inline code, all `@example`s should refer to a sample file and a tag within it.

nodejs-spanner has a great examples of this.  Here's one example ([link](https://github.com/googleapis/nodejs-spanner/blob/72efac10bcab961732cca234312908fa1b5bc3dd/src/batch-transaction.js#L116)):

```
/**
 ...
 * @example <caption>include:samples/batch.js</caption>
 * region_tag:spanner_batch_client
*/
```

### PR DESCRIPTION
**Pull request by vijay-qlogic**
Fixes #157 for table.js (it's a good idea to open an issue first for discussion)

- [ ] Tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

### CODE REVIEW

**Comment by sduskis** [comment]
@vijay-qlogic: Please fix the errors.

**Comment by sduskis** [comment]
@vijay-qlogic, please fix lint.

**Comment by sduskis** [comment]
@vijay-qlogic, can you please resolve the conflicts?

**Comment by sduskis** [comment]
The samples should be basically what the comments were.  They should not be full examples that include printlns.

File: samples/document-snippets/instance.js [hunk 1]
```diff
@@ -108,18 +103,18 @@ const snippets = {
 
     instance.createAppProfile(appProfileId, options, (err, appProfile) => {
       if (err) {
-        console.error('Error creating appProfile: ', err);
-        callback(err);
-        return;
+        // Handle the error.
+        return callback(err);
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Do we need a callback here?

File: samples/document-snippets/instance.js [hunk 2]
```diff
@@ -151,110 +146,108 @@ const snippets = {
...
     // [END bigtable_get_instance]
   },
 
-  getClusters: (instanceId, callback) => {
+  getClusters: instanceId => {
+    // [START bigtable_get_clusters]
+    const Bigtable = require('@google-cloud/bigtable');
+    const bigtable = new Bigtable();
     const instance = bigtable.instance(instanceId);
+    const caughtError = false;
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Do we need `caughtError`?

File: samples/document-snippets/instance.js [hunk 3]
```diff
@@ -151,110 +146,108 @@ const snippets = {
...
     // [END bigtable_get_clusters]
   },
 
-  getAppProfiles: (instanceId, callback) => {
+  getAppProfiles: instanceId => {
+    // [START bigtable_get_app_profiles]
+    const Bigtable = require('@google-cloud/bigtable');
+    const bigtable = new Bigtable();
     const instance = bigtable.instance(instanceId);
+    const caughtError = false;
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Do we need `caughtError`?

File: samples/document-snippets/instance.js [hunk 4]
```diff
@@ -151,110 +146,108 @@ const snippets = {
...
     // [END bigtable_get_app_profiles]
   },
 
-  getMetaData: (instanceId, callback) => {
+  getMetadata: instanceId => {
+    // [START bigtable_get_instance_metadata]
+    const Bigtable = require('@google-cloud/bigtable');
+    const bigtable = new Bigtable();
     const instance = bigtable.instance(instanceId);
+    const caughtError = false;
```
Comment (sduskis) [inline samples/document-snippets/instance.js]: Do we need `caughtError`?

Comment (vijay-qlogic) [reply samples/document-snippets/instance.js]: not in use, 'll remove it

**Pull request merged**